code/shared-mobility/shared-mobility-smrt.js

```diff
@@ -158,3 +158,210 @@ $(document).on("knack-scene-render.scene_339", function () {
 $(document).on("knack-scene-render.scene_340", function () {
   disableBreadCrumbsNonAdmin();
 });
+
+
+/******* Associate impoundments with invoice *******/
+
+function getTodaysDate() {
+  var today = new Date()
+  return((today.getMonth() + 1) + "/" + today.getDate() + "/"+ today.getFullYear())
+}
+
+var headers = {
+  "X-Knack-Application-ID": "6669fb3cd43ca60027942eef",
+  Authorization: Knack.getUserToken(),
+  "content-type": "application/json",
+};
+
+
+// Create invoice items from items after selection and submission
+function handleAddImpoundmentsClick(id, viewKey) {
+
+  var impoundmentDetailsView = "view_392";
+  var impoundmentDetailsScene = "scene_210";
+
+  // Get current invoice knack ID
+  var hrefArray = window.location.href.split("/");
+  var invoiceId = hrefArray[hrefArray.length - 2];
+
+  // Show spinner
+  $("#" + id).append(
+    '<span id="' +
+      id +
+      '-spinner" class="icon is-2x">&nbsp;<i class="fa fa-spinner fa-spin"></i></span>'
+  );
+
+  // Cycle through selected checkboxes, get their id and create payload
+
+  var selectedImpoundments = [];
+
+  $("#" + viewKey + " tbody input[type=checkbox]:checked").each(function () {
+    // Get id of row
+    var id = $(this).closest("tr").attr("id");
+
+     $.ajax({
+      type: "GET",
+      url:
+        "https://api.knack.com/v1/scenes/" + impoundmentDetailsScene + "/views/" +
+        impoundmentDetailsView + "/records/" + id,
+      headers: headers,
+      contentType: "application/json",
+      })
+        .then(function (res) {
+          newInvoiceItem = {
+              "field_404": res["field_121_raw"], // provider
+              "field_407": res["field_128_raw"], // impound fee: transaction amount
+              "field_454": res["field_115"], // description: date of impound
+              "field_403": getTodaysDate(), // Created Date
+              "field_401": "Impoundment Fee",
+              "field_408": [{ // Created By
+                  "id": Knack.user.id,
+                  "identifier": Knack.user.attributes.values.name.full
+                }],
+              "field_405": [{id:id}], // impoundment id
+              "field_411": "UNPAID", // transaction status
+              "crumbtrail": {"invoice-details_id": invoiceId}
+            }
+            return(newInvoiceItem)
+          })
+          .then(function(newInvoiceItem) {
+            $.ajax({
+              type: "POST",
+              url: "https://api.knack.com/v1/pages/scene_231/views/view_459/records",
+              headers: headers,
+              data: JSON.stringify(newInvoiceItem),
+              contentType: "application/json",
+            })
+            .then(function (res) {
+              Knack.views["view_459"].model.fetch();
+            })
+            .fail(function () {
```

Comment:
We saw this part fail during a demo in dev + product sync, but I can't replicate it today. [John was able to briefly](https://austininnovation.slack.com/archives/C026ZRP1TH8/p1723130889949649?thread_ts=1723128751.824889&cid=C026ZRP1TH8) but then it resolved.

We've used [code to show Knack-style error messages in the UI](https://github.com/cityofaustin/atd-knack/blob/70d9140ac60a0b7973fdf9615ca8f7cc43849ded/code/data-tracker/data-tracker.js#L497-L506) in the past. Might be a good addition to help users try again if we can't get to the bottom of the cause of the error.